mypy/stubgenc.py

```diff
@@ -254,10 +254,18 @@ def __init__(
                         "List",
                         "Literal",
                         "NamedTuple",
+                        "Never",
+                        "NoReturn",
+                        "Set",
                         "Optional",
                         "Tuple",
+                        "Type",
+                        "TypeIs",
+                        "TypeGuard",
                         "Union",
-                    ]
+                    ],
+                    "os": ["PathLike"],
+                    "pathlib": ["Path"],
```

Comment:
If it is easy enough please add a test for things imported from other modules than typing.